### ISSUE TITLE
`Deno.watchFs` with `recursive: false` still gives events for children.

### ISSUE THREAD

**Issue by jespertheend** [denoland/deno#18348]
Here is a reduced test case:
```ts
import {ensureFile} from "https://deno.land/std@0.180.0/fs/mod.ts";
import {resolve} from "https://deno.land/std@0.180.0/path/mod.ts";

const watchFilePath = "./foo/bar/baz.txt";
await ensureFile(watchFilePath);

const paths: Set<string> = new Set();
let lastPath = watchFilePath
paths.add(lastPath);
while (true) {
	lastPath = resolve(lastPath, "..");
	if (paths.has(lastPath)) break;
	paths.add(lastPath);
}

for (const path of paths) {
	createWatcher(path);
}

async function createWatcher(path: string) {
	const watcher = Deno.watchFs(path, {
		recursive: false,
	});
	for await (const e of watcher) {
		console.log(path, e.kind, e.paths);
	}
}

await Deno.writeTextFile("./foo/bar/unwatched.txt", String(Math.random()));
```
Run with `deno run -A watch.ts`

When running this I would expect no events to get logged, since a watcher for `unwatched.txt` was never created. But when I run this the following is logged instead:
```
/home/jesper/tmp/foo/bar modify [ "/home/jesper/tmp/foo/bar/unwatched.txt" ]
/home/jesper/tmp/foo/bar access [ "/home/jesper/tmp/foo/bar/unwatched.txt" ]
```

I'm on Ubuntu 22.10

**Comment by sigmaSd** [comment]
I'm not sure I follow, you have an active watcher for bar directory, I expect you get events for file written inside of it and that's what happens

**Comment by jespertheend** [comment]
@sigmaSd Yeah but the watcher was created with `recursive: false`, maybe I'm misunderstanding what that option does?
I would expect that option to filter out events from children. That's how it works when you only have a single watcher it seems.

**Comment by sigmaSd** [comment]
man inotify
```
    Inotify monitoring of directories is not recursive: to monitor subdirectories under a directory, additional watches must be  created.   This
       can take a significant amount time for large directory trees.
```

According to this it seems that recursive definition is monitoring sub-directories under the watch directory

**Comment by jespertheend** [comment]
That is the exact opposite of what I'm observing.
`Deno.watchFs("/path/to/dir")` gives me events for changes to any of the children in `/path/to/dir` and children in subdirectories.
`Deno.watchFs("/path/to/dir", {recursive: false})` only gives me events for changes to that directory, not children.

Only when multiple watchers are created does behavior start to get wonky.

**Comment by sigmaSd** [comment]
Its just that deno api calls inotify recursively by default

I still don't see the problem

**Comment by sigmaSd** [comment]
with a single watcher I still see the same (printing events for creating stuff under the watched dir) behavior

```ts
Deno.mkdirSync("q", { recursive: true });

(async () => {
  for await (
    const e of Deno.watchFs("q", {
      recursive: false,
    })
  ) {
    console.log(e);
  }
})();

await new Promise((r) => setTimeout(r, 500));

await Deno.writeTextFile("q/c", "");
```

**Comment by jespertheend** [comment]
Oh huh, that's strange, I could swear I wasn't able to reproduce this in an earlier test case with only a single watcher. I guess I must not have been paying attention.

With that out of the way though, what even is the point for the `recursive` flag then, if it has no effect on directories?

**Comment by greergan** [comment]
I started playing with Deno.watchFs today for the first time after doing a Deno upgrade and my watcher will only work on a single directory. It totally ignores recursive:true even though it is said to be the default.
